### PR TITLE
Add doc checker to CI

### PR DESCRIPTION
**Pull request by colinwahl**
**Description of the change**

Per suggestion by @thomashoneyman, sets up the [purescript-doc-check-action](https://github.com/colinwahl/purescript-doc-check-action) on this repository to automatically compile the docs found in the README (and any future docs in the `docs` directory. This is mostly a proof of concept for that action.

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a link to this PR and your username
- ~Linked any existing issues or proposals that this pull request should close~
- ~Updated or added relevant documentation in the README and/or documentation directory~
- ~Added a test for the contribution (if applicable)~

### CODE REVIEW

File: spago.dhall [hunk 1]
```diff
@@ -13,5 +13,5 @@
   , "unfoldable"
   ]
 , packages = ./packages.dhall
-, sources = [ "src/**/*.purs", "test/**/*.purs" ]
+, sources = [ "src/**/*.purs", "test/**/*.purs", "doc-test/*.purs" ]
```
Comment (thomashoneyman) [inline spago.dhall:16]: If we add code snippets to the docs directory will this pattern still pick it up (ie. it's just one flat directory) or do we need "doc-test/**/*.purs" to catch subdirectories?

File: README.md [hunk 2]
```diff
@@ -38,26 +36,22 @@ machine1 = do
...
 from the array. It then checks and halts on any zero input, and otherwise
 scales the inputs (by dividing by 2). The result is then transformed into a string.
 
-The resulting machine can be materialized via
-
-```purescript
-> toUnfoldable unit machine1 :: Array String
-["5","10","15","20","25"]
-```
+The resulting machine can be materialized via `toUnfoldable unit machine1 :: Array String`.
```
Comment (thomashoneyman) [inline README.md:48]: Maybe this could be moved to a comment above `result` in the code snippet instead of here.

**Comment by thomashoneyman** [comment]
Just as a quick question: if I want to add some pseudocode or something to a snippet, is it possible to opt-out of doc checking for that snippet? It's not planned for this library, but just curious in case we'd like to add this across the org.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -32,3 +32,6 @@ jobs:
 
       - name: Run tests
         run: spago test --no-install
+
+      - name: Check docs
+        uses: colinwahl/purescript-doc-check-action@main
```
Comment (thomashoneyman) [inline .github/workflows/ci.yml:37]: Do you anticipate any breaking changes to the tool / new required fields in the action.yml coming up that might break repositories downstream? If you're still iterating on this I'd just like to make sure CI doesn't break going forward.

**Comment by thomashoneyman** [comment]
This is a nice utility! I'm curious how @paluh and @vladciobanu, who worked on this library recently, feel about the decision for `import Prelude` to be implicit in these snippets. I'm fine with it, but I also understand wanting all the imports to be explicitly listed out.

I think this is worth adding to the CHANGELOG as an "other improvement", if you'd like. If you would rather not, that's also fine.

File: spago.dhall (hunk 1 above)
Comment (colinwahl) [reply spago.dhall:16]: This just creates a flat directory structure, so it'll be picked up just fine. I actually want to see if there is a way to not include this is the library code, but instead pass it in as an argument to `spago` at the callsite within `purescript-doc-check-action`

Comment (thomashoneyman) [reply spago.dhall:16]: Yes, that would be much better. I saw your experiment with the —path flag and hopefully that works.

**Comment by vladciobanu** [comment]
So each code block requires its own individual imports? That can prove tedious. I don't really mind automatically importing Prelude as much as having to add the import list to each code block.

Would it make more sense to export all blocks in a SINGLE module? That way, we could have a single comment block on top specifying the imports.

**Comment by paluh** [comment]
This is really cool improvement. Thanks a lot for working on it! I also really like that this can settle some easy to follow standard  how to organize this kind of testing.

From my perspective explicite imports are fine as they can be a useful part of the documentation. On the other hand it would be really great if we could somehow choose between test modules generation strategy (maybe if a code block starts with imports it should be treated as a separate module in other case as a continuation of the previous one). In many projects which I'm working on we have a single import block and transform everything into a single test module like here: [purescript-undefined-is-not-a-problem](https://github.com/paluh/purescript-undefined-is-not-a-problem/blob/master/README.md) or even in [purescript-postgresql-client](https://github.com/rightfold/purescript-postgresql-client/blob/master/README.md) or [purescript-selda](https://github.com/Kamirus/purescript-selda/blob/master/README.md).
We use [literate purescript lib](https://github.com/Thimoteus/literate-purescript) by @Thimoteus. We just put `pretest` command into the [`package.json`](https://github.com/paluh/purescript-undefined-is-not-a-problem/blob/master/package.json#L12). I know that there is [an another literate PS lib](https://github.com/paulyoung/literate-purescript) by @paulyoung. They are both probably good sources of inspiration or even starting points for a larger collaboration on this kind of a tool...

**Pull request closed without merging**